x-pack/legacy/plugins/code/server/usage_collector.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ServerFacade } from '../';
+import { APP_USAGE_TYPE } from '../common/constants';
+import { LanguageServerStatus } from '../common/language_server';
+import { CodeUsageMetrics } from '../model/usage_telemetry_metrics';
+import { EsClient } from './lib/esqueue';
+import { RepositoryObjectClient } from './search';
+import { LspService } from './lsp/lsp_service';
+import { CTAGS, GO, JAVA, TYPESCRIPT } from './lsp/language_servers';
+
+export async function fetchCodeUsageMetrics(client: EsClient, lspService: LspService) {
+  const repositoryObjectClient: RepositoryObjectClient = new RepositoryObjectClient(client);
+  const allRepos = await repositoryObjectClient.getAllRepositories();
+  const langServerEnabled = async (name: string) => {
+    const status = await lspService.languageServerStatus(name);
+    return status !== LanguageServerStatus.NOT_INSTALLED ? 1 : 0;
+  };
+  return {
+    [CodeUsageMetrics.ENABLED]: 1,
+    [CodeUsageMetrics.REPOSITORIES]: allRepos.length,
+    [CodeUsageMetrics.LANGUAGE_SERVERS]: {
```

Comment:
Can we convert this into key value array

```
[CodeUsageMetrics.LANGUAGE_SERVERS]: [{
   key: TYPESCRIPT.name,
   enabled: await langServerEnabled(TYPESCRIPT.name),
}, ...]
```

This way we dont have to update the telemetry es mapping on every language server addition.


Or even better if we can have this as an array of strings with only enabled languages this would be much easier for us to visualize and slice/dice it in kibana.


```
LANGUAGE_SERVERS_ENABLED = 'langServersEnabled';

...

[CodeUsageMetrics.LANGUAGE_SERVERS_ENABLED]: await (Promise.all([
  TYPESCRIPT.name,
  JAVA.name,
  CTAGS.name,
  GO.name,
].map(langServerEnabled))).filter(Boolean)
```

Reply:
This does make sense in terms of the metrics es mapping, not sure if this will affect the way we are going to aggregate these data. Anyway, will explore that bit when we have the initial data.